### PR TITLE
Move params implementation to new file

### PR DESCRIPTION
**Pull request by piponazo**
In this PR I am proposing to move the implementation of **Params**'s methods to its own .cpp file.

I started these changes just with the idea of implementing the existing Singleton in **Params** in the modern c++ way (see first commit). Then I realised that most of the file `exiv2.cpp` was the implementation of that **Params** class and I think it would be better to have it on its own file. Thanks to this change, now the `exiv2.cpp` file only has 90 LoC.

I recommend to review the PR commit by commit since I also took the opportunity to clang-format the exiv2app.* files.

### CODE REVIEW

**Comment by piponazo** [comment]
I was also thinking about renaming the files `exiv2app.*` to `Params.*`, to match the class name. What do you think about that?

File: src/exiv2app.cpp [hunk 1]
```diff
@@ -1278,9 +1278,7 @@ namespace
 
     bool parseCmdFiles(ModifyCmds& modifyCmds, const Params::CmdFiles& cmdFiles)
     {
-        Params::CmdFiles::const_iterator end = cmdFiles.end();
-        Params::CmdFiles::const_iterator filename = cmdFiles.begin();
-        for (; filename != end; ++filename) {
+        for (auto filename = cmdFiles.cbegin(); filename != cmdFiles.cend(); ++filename) {
```
Comment (D4N) [inline src/exiv2app.cpp]: Why not a range based for loop:
``` C++
for (const auto & filename : cmdFiles) {
  // stuff happens there
}
```

**Review by D4N — DISMISSED** [review]
Great cleanup, thank you!

File: src/exiv2app.cpp (hunk 1 above)
Comment (piponazo) [reply src/exiv2app.cpp]: Even better! I'll change it

**Comment by D4N** [comment]
> I was also thinking about renaming the files `exiv2app.*` to `Params.*`, to match the class name. What do you think about that?

Sorry, missed this comment: no objections from my side, I'm all in for more clarity.

**Review by D4N — APPROVED** [review]
LGTM!

**Pull request merged**